### PR TITLE
Address issues relating to creation of legacy migrations

### PR DESCRIPTION
**Pull request by Sybrand**
- Unit tests for edge cases.
- Making legacy serializer more permissive.
- Re-factoring stacker to allow for easier profiling.
- Some optimization.

### CODE REVIEW

File: app/backend/wells/stack.py [hunk 1]
```diff
@@ -239,19 +272,33 @@ def _stack(self, records, well: Well) -> Well:
...
         Used to update an existing well record.
         """
-        records = ActivitySubmission.objects.filter(well=submission.well)
+        # .select_related(
+        #         'casing_set',
+        #         'screen_set',
+        #         'linerperforation_set',
+        #         'decommission_description_set',
+        #         'lithologydescription_set'
+        #     ) \
```
Comment (stephenhillier) [inline app/backend/wells/stack.py]: commented code

File: app/backend/wells/stack.py [hunk 2]
```diff
@@ -199,9 +230,11 @@ def _stack(self, records, well: Well) -> Well:
                 submission.well_activity_type.code, WellStatusCode.types.other().well_status_code)
             source_target_map = activity_type_map.get(submission.well_activity_type.code, {})
             serializer = submissions.serializers.WellSubmissionStackerSerializer(submission)
+            # serializer.data.items is VERY slow - look at not using serializer when iterating through this.
             for source_key, value in serializer.data.items():
                 # We only consider items with values, and keys that are in our target
                 # an exception is STAFF_EDIT submissions (we need to be able to accept empty values)
+                # value = getattr(submission, source_key)
```
Comment (stephenhillier) [inline app/backend/wells/stack.py]:
```suggestion
```

File: app/backend/wells/stack.py [hunk 3]
```diff
@@ -99,15 +109,27 @@ def _create_legacy_submission(self, well: Well) -> None:
         data = {k: v for (k, v) in data.items() if v is not None and v != ''}
         # Retain the well reference.
         data['well'] = well.well_tag_number
+        # Clean out bad data.
+        # (There must be a way to deal with this on the serializer, but I can't figure it out!)
+        # owner_city = data.get('owner_city', None)
+        # if owner_city is not None and len(owner_city.strip()) == 0:
+        #   del data['owner_city']
```
Comment (stephenhillier) [inline app/backend/wells/stack.py]: commented code
